File: withJava/APPStokBarang.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.appstokbarang;

/**
 *
 * @author lenovo
 */

import java.util.ArrayList;
import java.util.Scanner;

public class APPStokBarang {
    static ArrayList<Barang> daftarBarang = new ArrayList<>();
    static Scanner input = new Scanner(System.in);

    public static void main(String[] args) {
        int pilihan;
        do {
            System.out.println("\n=== APLIKASI STOK BARANG ===");
            System.out.println("1. Tambah Barang Baru");
            System.out.println("2. Hapus Barang");
            System.out.println("3. Tambah Stok Barang");
            System.out.println("4. Kurangi Stok Barang");
            System.out.println("5. Tampilkan Daftar Barang");
            System.out.println("0. Keluar");
            System.out.print("Pilih menu: ");
            pilihan = input.nextInt();
            input.nextLine(); // membersihkan buffer

            switch (pilihan) {
                case 1 -> tambahBarang();
                case 2 -> hapusBarang();
                case 3 -> tambahStok();
                case 4 -> kurangiStok();
                case 5 -> tampilkanBarang();
                case 0 -> System.out.println("Keluar dari program...");
                default -> System.out.println("Pilihan tidak valid!");
            }
        } while (pilihan != 0);
    }

    static void tambahBarang() {
        System.out.print("Masukkan kode barang: ");
        String kode = input.nextLine();
        System.out.print("Masukkan nama barang: ");
        String nama = input.nextLine();
        System.out.print("Masukkan jumlah stok awal: ");
        int stok = input.nextInt();

        daftarBarang.add(new Barang(kode, nama, stok));
        System.out.println("Barang berhasil ditambahkan!");
    }

    static void hapusBarang() {
        System.out.print("Masukkan kode barang yang akan dihapus: ");
        String kode = input.nextLine();
        boolean ditemukan = false;

        for (int i = 0; i < daftarBarang.size(); i++) {
            if (daftarBarang.get(i).getKode().equals(kode)) {
                daftarBarang.remove(i);
                System.out.println("Barang berhasil dihapus!");
                ditemukan = true;
                break;
            }
        }

        if (!ditemukan) {
            System.out.println("Barang tidak ditemukan!");
        }
    }

    static void tambahStok() {
        System.out.print("Masukkan kode barang: ");
        String kode = input.nextLine();
        for (Barang b : daftarBarang) {
            if (b.getKode().equals(kode)) {
                System.out.print("Masukkan jumlah yang akan ditambah: ");
                int jumlah = input.nextInt();
                b.setStok(b.getStok() + jumlah);
                System.out.println("Stok berhasil ditambahkan!");
                return;
            }
        }
        System.out.println("Barang tidak ditemukan!");
    }

    static void kurangiStok() {
        System.out.print("Masukkan kode barang: ");
        String kode = input.nextLine();
        for (Barang b : daftarBarang) {
            if (b.getKode().equals(kode)) {
                System.out.print("Masukkan jumlah yang akan dikurangi: ");
                int jumlah = input.nextInt();
                if (jumlah <= b.getStok()) {
                    b.setStok(b.getStok() - jumlah);
                    System.out.println("Stok berhasil dikurangi!");
                } else {
                    System.out.println("Jumlah melebihi stok yang tersedia!");
                }
                return;
            }
        }
        System.out.println("Barang tidak ditemukan!");
    }

    static void tampilkanBarang() {
        if (daftarBarang.isEmpty()) {
            System.out.println("Belum ada data barang!");
        } else {
            System.out.println("\n=== DAFTAR STOK BARANG ===");
            for (Barang b : daftarBarang) {
                System.out.println("Kode: " + b.getKode() + " | Nama: " + b.getNama() + " | Stok: " + b.getStok());
            }
        }
    }
}

File: withPython/APPStokBarang.py
from Barang import Barang

class APPStokBarang:
    def __init__(self):
        self.daftar_barang = []

    def tambah_barang(self):
        kode = input("Masukkan kode barang: ")
        nama = input("Masukkan nama barang: ")
        jumlah = int(input("Masukkan jumlah awal: "))
        barang = Barang(kode, nama, jumlah)
        self.daftar_barang.append(barang)
        print("✅ Barang berhasil ditambahkan!\n")

    def hapus_barang(self):
        kode = input("Masukkan kode barang yang ingin dihapus: ")
        for barang in self.daftar_barang:
            if barang.kode == kode:
                self.daftar_barang.remove(barang)
                print("🗑️ Barang berhasil dihapus!\n")
                return
        print("❌ Barang tidak ditemukan!\n")

    def tambah_stok(self):
        kode = input("Masukkan kode barang: ")
        jumlah = int(input("Masukkan jumlah yang ingin ditambah: "))
        for barang in self.daftar_barang:
            if barang.kode == kode:
                barang.tambah_stok(jumlah)
                print("✅ Stok berhasil ditambahkan!\n")
                return
        print("❌ Barang tidak ditemukan!\n")

    def kurangi_stok(self):
        kode = input("Masukkan kode barang: ")
        jumlah = int(input("Masukkan jumlah yang ingin dikurangi: "))
        for barang in self.daftar_barang:
            if barang.kode == kode:
                barang.kurangi_stok(jumlah)
                print("✅ Stok berhasil dikurangi!\n")
                return
        print("❌ Barang tidak ditemukan!\n")

    def tampilkan_stok(self):
        if not self.daftar_barang:
            print("📦 Belum ada barang di stok.\n")
        else:
            print("\n=== DAFTAR STOK BARANG ===")
            for barang in self.daftar_barang:
                print(barang)
            print()

    def menu(self):
        while True:
            print("==== APLIKASI STOK BARANG ====")
            print("1. Tambah Barang Baru")
            print("2. Hapus Barang")
            print("3. Tambah Stok")
            print("4. Kurangi Stok")
            print("5. Tampilkan Semua Barang")
            print("6. Keluar")

            pilihan = input("Pilih menu (1-6): ")
            print()

            if pilihan == "1":
                self.tambah_barang()
            elif pilihan == "2":
                self.hapus_barang()
            elif pilihan == "3":
                self.tambah_stok()
            elif pilihan == "4":
                self.kurangi_stok()
            elif pilihan == "5":
                self.tampilkan_stok()
            elif pilihan == "6":
                print("Terima kasih! Program selesai.")
                break
            else:
                print("❌ Pilihan tidak valid!\n")


if __name__ == "__main__":
    app = APPStokBarang()
    app.menu()
